### ISSUE TITLE
TestingRecaptchaValidator does not implement all methods from RecaptchaValidator

### ISSUE THREAD

**Issue by ivansenic** [mkopylec/recaptcha-spring-boot-starter#16]
We are hitting `NullPointerException` in our tests with `recaptcha.testing.enabled=true`. After some debugging we figured out that `TestRecaptchaValidator` is not implementing `validate(String userResponse, HttpServletRequest request)` method. Since it's extending the `RecaptchaValidator` we end-up there and hit the NPE as the `ipAddressResolver` is `null`.

Fortunately this is avoidable in our tests as we can define our own extended `TestRecaptchaValidator` instance.
 
I guess this is an easy fix in fact.. But to avoid this problem in future I would advise to create a `IRecaptchaValidator` interface and remove inheritance in TestRecaptchaValidator. That's a simple refactoring task in the IDE.

```
java.lang.NullPointerException: null
	at com.github.mkopylec.recaptcha.validation.RecaptchaValidator.validate(RecaptchaValidator.java:42) ~[recaptcha-spring-boot-starter-2.3.0.jar:na]
	at com.github.mkopylec.recaptcha.security.RecaptchaAuthenticationFilter.attemptAuthentication(RecaptchaAuthenticationFilter.java:52) ~[recaptcha-spring-boot-starter-2.3.0.jar:na]
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:212) ~[spring-security-web-5.1.1.RELEASE.jar:5.1.1.RELEASE]
```

But besides this we have no other problems and in production mode everything works, even with reCAPTCHA v3.

Cheers,
Ivan

**Comment by mkopylec** [comment]
Hi, thanks for reporting the bug.
I'll fix it in the next release.

**Comment by mkopylec** [comment]
Fixed in 2.3.1
